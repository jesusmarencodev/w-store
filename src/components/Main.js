import React from 'react';
import Head from './Head';
import { Container } from './Container';

const Main = () => {
  return (
    <div className="main">
      <Head />
      <h3 className="main__title">Store</h3>
      <Container />
    </div>
  );
};

export default Main;
