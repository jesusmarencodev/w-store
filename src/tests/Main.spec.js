import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Main from '../components/Main';

const mockStore = configureStore([]);

describe('Main component', () => {
  it('Snapshot', () => {
    const initialState = {
      products: {
        productList: [],
        shoppingCar: false,
      },
    };

    const store = mockStore(initialState);

    const component = renderer.create(
      <Provider store={store}>
        <Main />
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
