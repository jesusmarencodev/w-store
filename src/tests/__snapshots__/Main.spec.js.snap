// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component Snapshot 1`] = `
<div
  className="main"
>
  <header
    className="head d-flex a-center"
  >
    <div
      className="head__logo"
    >
      <img
        alt="store"
        className="head__img"
        src="store.png"
      />
    </div>
    <div
      className="head__shop d-flex false"
    >
      <div
        className="shop__button"
        onClick={[Function]}
      >
        <img
          alt="store"
          className="head__img-car"
          src="car.png"
        />
        <span
          className="shop-total false"
        >
          $
          0
        </span>
      </div>
    </div>
  </header>
  <h3
    className="main__title"
  >
    Store
  </h3>
  <div
    className="d-flex container"
  >
    <article
      className="listProducts"
    >
      <div
        className="shadow"
      />
      <div
        className="scrollbar"
      >
        <div
          className="listProducts__list"
        />
      </div>
    </article>
    <aside
      className="carShopping"
    >
      <div
        className="shadow"
        style={
          Object {
            "marginRight": "2.5rem",
          }
        }
      />
      <div
        className="scrollbar"
      >
        <div>
          <p
            className="product__description"
            style={
              Object {
                "paddingTop": "2.5rem",
              }
            }
          >
            Please choose a product on the left.
          </p>
        </div>
      </div>
    </aside>
  </div>
</div>
`;
